SU2_CFD/include/objectives/turbomachinery/CTurbomachineryPerformance.hpp

```diff
@@ -0,0 +1,256 @@
+  CFluidModel &FluidModel;
+
+  public:
+  CTurbomachineryBladePerformance(CFluidModel& fluidModel,
+                                  unsigned short nDim,
+                                  su2double areaIn,
+                                  su2double radiusIn,
+                                  su2double areaOut,
+                                  su2double radiusOut);
+
+  virtual void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) {};
+
+  const CTurbomachineryState& GetInletState() { return InletState; }
+
+  const CTurbomachineryState& GetOutletState()  { return OutletState; }
+
+  su2double GetKineticEnergyLoss() const { return KineticEnergyLoss; }
+
+  su2double GetTotalPressureLoss() const { return TotalPressureLoss; }
+
+  su2double GetEntropyGen() const { return EntropyGen; }
+
+  su2double GetPressureRatio() const { return PressureRatio; }
+
+  su2double GetEulerianWork() const { return EulerianWork; }
+
+};
+
+class CTurbineBladePerformance : public CTurbomachineryBladePerformance {
+
+  public:
+  CTurbineBladePerformance(CFluidModel& fluidModel,
+                           unsigned short nDim,
+                           su2double areaIn,
+                           su2double radiusIn,
+                           su2double areaOut,
+                           su2double radiusOut);
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+
+class CCompressorBladePerformance : public CTurbomachineryBladePerformance {
+
+  public:
+  CCompressorBladePerformance(CFluidModel& fluidModel,
+                              unsigned short nDim,
+                              su2double areaIn,
+                              su2double radiusIn,
+                              su2double areaOut,
+                              su2double radiusOut);
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+
+class CPropellorBladePerformance : public CTurbomachineryBladePerformance {
+
+  public:
+  CPropellorBladePerformance(CFluidModel& fluidModel,
+                              unsigned short nDim,
+                              su2double areaIn,
+                              su2double radiusIn,
+                              su2double areaOut,
+                              su2double radiusOut);
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+
+class CTurbomachineryStagePerformance {
+   protected: 
+      su2double TotalStaticEfficiency, TotalTotalEfficiency, KineticEnergyLoss, TotalPressureLoss, EntropyGen, PressureRatio, EulerianWork;
+      CFluidModel &fluidModel;
+   public:
+      CTurbomachineryStagePerformance(CFluidModel& fluid);
+      ~CTurbomachineryStagePerformance();
+      virtual void ComputePerformanceStage(CTurbomachineryState InState, CTurbomachineryState OutState, CConfig* config);
```

Comment:
```suggestion
      virtual ~CTurbomachineryStagePerformance() = default;
      virtual void ComputePerformanceStage(CTurbomachineryState InState, CTurbomachineryState OutState, const CConfig* config);
```
If you have virtual functions you need a virtual destructor.
Can the config be const?